pkgs/servers/klipper/klipper-flash.nix

```diff
@@ -0,0 +1,41 @@
+{ lib
+, writeShellApplication
+, gnumake
+, pkgsCross
+, klipper
+, klipper-firmware
+, python2
+, avrdude
+, stm32flash
+, mcu ? "mcu"
+, flashDevice ? "/dev/null"
+, firmwareConfig ? ./simulator.cfg
+}:
+let
+  isNotSupported = with builtins; isNull (match ''^.*CONFIG_BOARD_DIRECTORY="(avr|stm32|lpc176x)".*$'' (readFile firmwareConfig));
+  isNotStm = with builtins; isNull (match ''^.*CONFIG_BOARD_DIRECTORY="(stm32)".*$'' (readFile firmwareConfig));
+in
+writeShellApplication {
+  name = "klipper-flash-${mcu}";
+  runtimeInputs = [
+    python2
```

Comment:
What causes this dep? Py2 is EOL.

Reply:
Yeah it's unfortunate, but klipper is still using Python2 largely 😅
https://github.com/Klipper3d/klipper/search?p=1&q=python2

All the Makefiles, flashing python files and so on all use python2. I believe there is an ongoing attempt at klipper to rewrite everything to python3. The current `klippy` derivation also has python2.